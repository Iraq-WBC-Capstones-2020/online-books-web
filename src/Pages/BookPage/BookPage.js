import React, { useEffect, useState } from 'react';
import './style.scss';
import BookPageSection from '../../components/BookPage/BookPage';
import Book from '../../models/Book';
import Spinner from '../../components/Spinner/Spinner';
import { useParams } from 'react-router-dom';
function BookPage() {
  const { bookId } = useParams();
  const [book, setBook] = useState();
  useEffect(() => {
    (async () => {
      const book = await Book.getBookFromDB(bookId);
      setBook(book);
    })();
  }, [bookId]);
  return <>{!book ? <Spinner /> : <BookPageSection book={book} />}</>;
}

export default BookPage;
