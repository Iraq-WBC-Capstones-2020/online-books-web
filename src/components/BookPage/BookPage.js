import React from 'react';
import { useTranslation } from 'react-i18next';
import CreateStars from '../CarsouelSlick/CreateStars';
import Paragraph from '../Paragraph/Paragraph';
import { MDBContainer, MDBRow, MDBCol, MDBBtn, MDBIcon } from 'mdbreact';
import { useSelector } from 'react-redux';
function BookPage({ book }) {
  const { t } = useTranslation();
  const {
    id,
    name,
    type,
    price,
    dis_count,
    cover,
    have_audio,
    description,
    audio_ref,
    author_id,
    author_name,
    released_date,
    views,
    rating,
    ref,
    tags,
  } = book;
  const user = useSelector((state) => state.user);

  const isLoggedIn = user.isSigned;
  const yearReleased = released_date.toDate().getFullYear();
  return (
    <MDBContainer className="mt-5">
      <MDBRow>
        <MDBCol sm="6" size="12" className="order-1 order-sm-0">
          <div className="mt-5">
            <h1 className="blue-text"> {name}</h1>
            <span className="gray-text">{author_name}</span>
            <br />
            <div className="mt-2 yellow-star">
              <CreateStars rate={rating} />
            </div>

            <Paragraph>{description}</Paragraph>
            <span className="gray-text">{t('bookPage.language')}:Arabic</span>
          </div>
          <br />
          {isLoggedIn ? (
            <>
              <div className="mt-3">
                <MDBBtn color="info" className="mr-3 px-3 py-0">
                  {tags}
                </MDBBtn>
                <MDBBtn color="info" className="px-3 py-0">
                  {yearReleased}
                </MDBBtn>
              </div>
              <div className="mt-4">
                <MDBIcon
                  outline
                  className="rounded-circle"
                  color="black"
                  icon="headphones-alt"
                  size="4x"
                />
                <MDBIcon className="ml-5" icon="arrow-down" size="4x" />
              </div>
            </>
          ) : (
            <>
              <span className="gray-text">
                Price: <strong>{price}IQD</strong>
              </span>

              <div className="mt-3">
                <MDBBtn color="info" className="mr-3 px-3 py-0">
                  {tags}
                </MDBBtn>
                <MDBBtn color="info" className="px-3 py-0">
                  {yearReleased}
                </MDBBtn>
              </div>
              <div className="mt-4">
                <MDBBtn color="primary" className="px-4 rounded-pill">
                  {t('bookPage.addToCart')}
                </MDBBtn>
              </div>
            </>
          )}
        </MDBCol>

        <MDBCol sm="6" size="12" className="order-0 order-sm-1">
          <div className="mt-5 d-flex justify-content-center align-self-center">
            <img src={cover} alt="Book cover" width="353" height="498" />
          </div>
        </MDBCol>
      </MDBRow>
    </MDBContainer>
  );
}

export default BookPage;
