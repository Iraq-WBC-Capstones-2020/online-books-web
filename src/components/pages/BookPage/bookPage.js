import React from 'react';
import './style.scss';
import CreateStars from './starRate';
import { MDBContainer, MDBRow, MDBCol, MDBBtn, MDBIcon } from 'mdbreact';
const BookPage = () => {
  const bookPage = {
    bookName: 'Mind Without Body',
    bookCover: 'https://via.placeholder.com/353x498',
    description: `is simply dummy text of the printing and typesetting industry.
    Lorem Ipsum has been the industry's standard dummy text ever since
    the 1500s, when an unknown printer took a galley of type and
    scrambled it to make a type specimen book.galley of type and
    scrambled it to make a type specimen book`,
    authortName: 'Ahmed khaled tofeq',
    bookType: 'Fantasy',
    publishDate: '2017',
    price: '15$',
  };

  const isLoggedIn = false;
  return (
    <MDBContainer className="mt-5">
      <MDBRow>
        <MDBCol sm="6" size="12">
          <div className="mt-5">
            <h1 className="blue-text"> {bookPage.bookName}</h1>
            <span className="gray-text">{bookPage.authortName}</span>
            <br />
            <div className="mt-2 yellow-star">
              <CreateStars />
            </div>
            <p className="mt-3 text-2xl leading-7 text-justify">
              {bookPage.description}
            </p>
            <span className="gray-text">language:Arabic</span>
          </div>
          <br />
          {isLoggedIn ? (
            <>
              <div className="mt-3">
                <MDBBtn color="info" className="mr-3 px-3 py-0">
                  {bookPage.bookType}
                </MDBBtn>
                <MDBBtn color="info" className="px-3 py-0">
                  {bookPage.publishDate}
                </MDBBtn>
              </div>
              <div className="mt-4">
                <MDBIcon
                  outline
                  className="rounded-circle"
                  color="black"
                  icon="headphones-alt"
                  size="4x"
                />
                <MDBIcon className="ml-5" icon="arrow-down" size="4x" />
              </div>
            </>
          ) : (
            <>
              <span className="gray-text">
                Price: <strong>{bookPage.price}</strong>
              </span>

              <div className="mt-3">
                <MDBBtn color="info" className="mr-3 px-3 py-0">
                  {bookPage.bookType}
                </MDBBtn>
                <MDBBtn color="info" className="px-3 py-0">
                  {bookPage.publishDate}
                </MDBBtn>
              </div>
              <div className="mt-4">
                <MDBBtn color="primary" className="px-4 rounded-pill">
                  addToCart
                </MDBBtn>
              </div>
            </>
          )}
        </MDBCol>

        <MDBCol sm="6" size="12">
          <div className="mt-5 d-flex justify-content-center align-self-center">
            <img
              src={bookPage.bookCover}
              alt="Book cover"
              width="353"
              height="498"
            />
          </div>
        </MDBCol>
      </MDBRow>
    </MDBContainer>
  );
};

export default BookPage;
