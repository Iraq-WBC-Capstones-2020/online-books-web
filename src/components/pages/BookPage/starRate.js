import React from 'react';
import PropTypes from 'prop-types';
function CreateStars({ rate }) {
  let stars = [];
  let firstDigit = rate - parseInt(rate);
  let secondDigit = firstDigit.toString().split('.')[1];
  let createStars = () => {
    for (let i = 1; i <= 5; i++) {
      if (i <= rate) {
        stars.push(<icon className="fa fa-star text-yellow-300 mr-1 "></icon>);
      } else {
        if (secondDigit === 5) {
          secondDigit = 0;
          stars.push(
            <icon className="fas fa-star-half-alt text-yellow-300 mr-1 "></icon>
          );
        } else {
          stars.push(
            <icon className="fa fa-star text-yellow-300 mr-1 font-light "></icon>
          );
        }
      }
    }
    return stars;
  };
  return <>{createStars()}</>;
}
CreateStars.propTypes = {
  rate: PropTypes.string,
};
export default CreateStars;
