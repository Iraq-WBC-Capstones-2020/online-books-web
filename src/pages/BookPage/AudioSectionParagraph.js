import React from 'react';
import './style.css';
import bookCover from './bookCover.jpg';
import { MDBContainer, MDBRow, MDBCol, MDBBtn, MDBIcon } from 'mdbreact';
const AudioSection = () => {
  return (
    <MDBContainer className="mt-5">
      <MDBRow>
        <MDBCol sm="6" size="12">
          <div className="mt-5 ">
            <img src={bookCover} alt="Book cover" width="353" height="498" />
          </div>
        </MDBCol>
        <MDBCol sm="6" size="12">
          <div className="mt-5 ">
            <h1 className="blue-text"> Mind without body</h1>
            <p className="mt-3  paragraph">
              is simply dummy text of the printing and typesetting industry.
              Lorem Ipsum has been the industry's standard dummy text ever since
              the 1500s, when an unknown printer took a galley of type and
              scrambled it to make a type specimen book.
            </p>
          </div>
        </MDBCol>
      </MDBRow>
    </MDBContainer>
  );
};

export default AudioSection;
