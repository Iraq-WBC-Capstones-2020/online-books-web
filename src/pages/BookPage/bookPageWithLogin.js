import React from 'react';
import './style.scss';
import bookCover from './bookCover.jpg';
import { MDBContainer, MDBRow, MDBCol, MDBBtn, MDBIcon } from 'mdbreact';
import { useTranslation } from 'react-i18next';
const BlogPage = () => {
  const { t } = useTranslation();
  const bookName = 'Mind Without Body';
  return (
    <MDBContainer className="mt-5">
      <MDBRow>
        <MDBCol sm="6" size="12">
          <div className="mt-5">
            <h1 className="blue-text">{bookName}</h1>
            <span className="gray-text">Ahmed khaled tofeq</span>
            <br />
            <div className="mt-2 yellow-star">
              <MDBIcon icon="star"></MDBIcon>
              <MDBIcon icon="star"></MDBIcon>
              <MDBIcon icon="star"></MDBIcon>
              <MDBIcon icon="star"></MDBIcon>
              <MDBIcon icon="star"></MDBIcon>
            </div>
            <p className="mt-3 paragraph">
              is simply dummy text of the printing and typesetting industry.
              Lorem Ipsum has been the industry's standard dummy text ever since
              the 1500s, when an unknown printer took a galley of type and
              scrambled it to make a type specimen book.galley of type and
              scrambled it to make a type specimen book.
            </p>
            <span className="gray-text">{`${t('language')}:Arabic`}</span>
            <br />
          </div>
          <div className="mt-3">
            <MDBBtn color="info" className="mr-3 px-3 py-0">
              Fantasy
            </MDBBtn>
            <MDBBtn color="info" className=" px-3 py-0">
              2017
            </MDBBtn>
          </div>
          <div className="mt-4">
            <MDBIcon
              outline
              className="  rounded-circle"
              color="black"
              icon="headphones-alt"
              size="4x"
            />
            <MDBIcon className="ml-5" icon="arrow-down" size="4x" />
          </div>
        </MDBCol>

        <MDBCol sm="6" size="12">
          <div className="mt-5 d-flex justify-content-center align-self-center ">
            <img src={bookCover} alt="Book cover" width="353" height="498" />
          </div>
        </MDBCol>
      </MDBRow>
    </MDBContainer>
  );
};

export default BlogPage;
