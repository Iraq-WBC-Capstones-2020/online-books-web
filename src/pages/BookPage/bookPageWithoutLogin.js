import React from 'react';
import './style.scss';
import bookCover from './bookCover.jpg';
import { MDBContainer, MDBRow, MDBCol, MDBBtn, MDBIcon } from 'mdbreact';
import { useTranslation } from 'react-i18next';
const BookPage = () => {
  const { t } = useTranslation();
  const bookName = 'Mind Without Body';
  const authortName = 'Ahmed khaled tofeq';
  const bookType = 'Fantasy';
  const publishDate = '2017';
  const isLoggedIn = false;
  return (
    <MDBContainer className="mt-5">
      <MDBRow>
        <MDBCol sm="6" size="12">
          <div className="mt-5">
            <h1 className="blue-text"> {bookName}</h1>
            <span className="gray-text">{authortName}</span>
            <br />
            <div className="mt-2 yellow-star">
              <MDBIcon icon="star"></MDBIcon>
              <MDBIcon icon="star"></MDBIcon>
              <MDBIcon icon="star"></MDBIcon>
              <MDBIcon icon="star"></MDBIcon>
              <MDBIcon icon="star"></MDBIcon>
            </div>
            <p className="mt-3 paragraph">
              {`is simply dummy text of the printing and typesetting industry.
              Lorem Ipsum has been the industry's standard dummy text ever since
              the 1500s, when an unknown printer took a galley of type and
              scrambled it to make a type specimen book.galley of type and
              scrambled it to make a type specimen book.`}
            </p>
            <span className="gray-text">{t('language')}:Arabic</span>
          </div>
          <br />
          {!isLoggedIn && (
            <>
              <span className="gray-text">
                Price: <strong>15$</strong>
              </span>

              <div className="mt-3">
                <MDBBtn color="info" className="mr-3 px-3 py-0">
                  {bookType}
                </MDBBtn>
                <MDBBtn color="info" className="px-3 py-0">
                  {publishDate}
                </MDBBtn>
              </div>
              <div className="mt-4">
                <MDBBtn color="primary" className="px-4 rounded-pill">
                  {t('addToCart')}
                </MDBBtn>
              </div>
            </>
          )}
          {isLoggedIn && (
            <>
              <div className="mt-3">
                <MDBBtn color="info" className="mr-3 px-3 py-0">
                  {bookType}
                </MDBBtn>
                <MDBBtn color="info" className="px-3 py-0">
                  {publishDate}
                </MDBBtn>
              </div>
              <div className="mt-4">
                <MDBIcon
                  outline
                  className="rounded-circle"
                  color="black"
                  icon="headphones-alt"
                  size="4x"
                />
                <MDBIcon className="ml-5" icon="arrow-down" size="4x" />
              </div>
            </>
          )}
        </MDBCol>

        <MDBCol sm="6" size="12">
          <div className="mt-5 d-flex justify-content-center align-self-center">
            <img src={bookCover} alt="Book cover" width="353" height="498" />
          </div>
        </MDBCol>
      </MDBRow>
    </MDBContainer>
  );
};

export default BookPage;
