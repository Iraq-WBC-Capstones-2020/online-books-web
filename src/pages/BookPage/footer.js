import React from 'react';
import { MDBContainer, MDBRow, MDBCol } from 'mdbreact';

const footer = () => {
  return (
    <MDBContainer className="mt-5">
      <MDBRow>
        <MDBCol size="12" sm="4">
          <div>
            <img src="" />
            <p>
              Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit
              aut fugit, sed quia consequuntur mag
            </p>
            <span>Copyright</span>
          </div>
        </MDBCol>
        <MDBCol size="12" sm="4">
          <h4 className="blue-text">Follow us</h4>
        </MDBCol>
        <MDBCol size="12" sm="4">
          <h4 className="blue-text">Tags</h4>
        </MDBCol>
      </MDBRow>
    </MDBContainer>
  );
};

export default footer;
