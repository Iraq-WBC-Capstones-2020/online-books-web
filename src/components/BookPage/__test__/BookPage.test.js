import React from 'react';
import renderer from 'react-test-renderer';
import BookPage from './../Details';

test('BookPage snapshot', () => {
  const tree = renderer.create(<BookPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
