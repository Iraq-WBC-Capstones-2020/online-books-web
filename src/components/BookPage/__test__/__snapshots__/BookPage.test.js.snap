// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookPage snapshot 1`] = `
<div
  className="container mt-5"
  data-test="container"
>
  <div
    className="row"
    data-test="row"
  >
    <div
      className="col-12 col-sm-6 order-1 order-sm-0"
      data-test="col"
    >
      <div
        className="mt-5"
      >
        <h1
          className="blue-text"
        >
           
          Mind Without Body
        </h1>
        <span
          className="gray-text"
        >
          Ahmed khaled tofeq
        </span>
        <br />
        <div
          className="mt-2 yellow-star"
        >
          <icon
            className="fa fa-star fa-md mr-1 text-yellow-300"
          />
          <icon
            className="fa fa-star fa-md mr-1 text-yellow-300"
          />
          <icon
            className="fa fa-star fa-md mr-1 text-yellow-300"
          />
          <icon
            className="fas fa-star-half-alt fa-md mr-1 text-yellow-300"
          />
          <icon
            className=" fa fa-star font-light fa-md mr-1 text-yellow-300"
          />
        </div>
        <p
          className="text-3xl font-medium border-b-4 inline border-blue-500"
        />
        <p
          className="text-base leading-loose text-justify"
        >
          is simply dummy text of the printing and typesetting industry.
    Lorem Ipsum has been the industry's standard dummy text ever since
    the 1500s, when an unknown printer took a galley of type and
    scrambled it to make a type specimen book.galley of type and
    scrambled it to make a type specimen book
        </p>
        <span
          className="gray-text"
        >
          bookPage.language
          :Arabic
        </span>
      </div>
      <br />
      <span
        className="gray-text"
      >
        Price: 
        <strong>
          15$
        </strong>
      </span>
      <div
        className="mt-3"
      >
        <button
          className="btn-info btn Ripple-parent mr-3 px-3 py-0"
          data-test="button"
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          Fantasy
          <div
            className="Ripple "
            data-test="waves"
            style={
              Object {
                "height": "0px",
                "left": "0px",
                "top": "0px",
                "width": "0px",
              }
            }
          />
        </button>
        <button
          className="btn-info btn Ripple-parent px-3 py-0"
          data-test="button"
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          2017
          <div
            className="Ripple "
            data-test="waves"
            style={
              Object {
                "height": "0px",
                "left": "0px",
                "top": "0px",
                "width": "0px",
              }
            }
          />
        </button>
      </div>
      <div
        className="mt-4"
      >
        <button
          className="btn-primary btn Ripple-parent px-4 rounded-pill"
          data-test="button"
          onMouseUp={[Function]}
          onTouchStart={[Function]}
          type="button"
        >
          bookPage.addToCart
          <div
            className="Ripple "
            data-test="waves"
            style={
              Object {
                "height": "0px",
                "left": "0px",
                "top": "0px",
                "width": "0px",
              }
            }
          />
        </button>
      </div>
    </div>
    <div
      className="col-12 col-sm-6 order-0 order-sm-1"
      data-test="col"
    >
      <div
        className="mt-5 d-flex justify-content-center align-self-center"
      >
        <img
          alt="Book cover"
          height="498"
          src="https://via.placeholder.com/353x498"
          width="353"
        />
      </div>
    </div>
  </div>
</div>
`;
